packages/teleport/src/services/nodes/nodes.ts

```diff
@@ -28,7 +30,18 @@ const service = {
   },
 
   createNodeJoinToken() {
-    return api.post(cfg.getNodeJoinTokenUrl()).then(makeNodeJoinToken);
+    return api.post(cfg.getNodeJoinTokenUrl()).then(makeNodeToken);
+  },
+
+  createNodeBashCommand() {
+    return this.createNodeJoinToken().then(makeNodeBashCmd);
+  },
+
+  createAppBashCommand(appName: string, appUri: string) {
+    return api
+      .post(cfg.getNodeJoinTokenUrl())
+      .then(makeNodeToken)
+      .then(token => makeAppBashCmd(token, appName, appUri));
```

Comment:
url

Reply:
Having URI in the code is fine. If you noticed, the query parameter in the URL named `uri` as this is a convention. There is one thing to limit the options of URI to the user and another to start renaming variable names. From technical perspective url is URI this is why I did not rename it in the code.